libs/feature/editor/src/lib/components/import-record/import-record.component.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { ImportRecordComponent } from './import-record.component'
+import { ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core'
+import { Router } from '@angular/router'
+import { TranslateModule, TranslateService } from '@ngx-translate/core'
+import { NotificationsService } from '@geonetwork-ui/feature/notifications'
+import { RecordsRepositoryInterface } from '@geonetwork-ui/common/domain/repository/records-repository.interface'
+import { of, throwError } from 'rxjs'
+
+class MockRouter {
+  navigate = jest.fn().mockReturnValue(Promise.resolve(true))
+}
+
+class MockTranslateService {
+  instant = jest.fn((key: string) => key)
+}
+
+class MockNotificationsService {
+  showNotification = jest.fn()
+}
+
+class MockRecordsRepository {
+  importRecordFromExternalFileUrlAsDraft = jest.fn(() => of(''))
+}
+
+class MockChangeDetectorRef {
+  markForCheck = jest.fn()
+}
+
+describe('ImportRecordComponent', () => {
+  let component: ImportRecordComponent
+  let fixture: ComponentFixture<ImportRecordComponent>
+  let mockRouter: MockRouter
+  let mockTranslateService: MockTranslateService
+  let mockNotificationsService: MockNotificationsService
+  let mockRecordsRepository: MockRecordsRepository
+  let mockChangeDetectorRef: Partial<ChangeDetectorRef>
+
+  beforeEach(async () => {
+    mockRouter = new MockRouter()
+    mockTranslateService = new MockTranslateService()
+    mockNotificationsService = new MockNotificationsService()
+    mockRecordsRepository = new MockRecordsRepository()
+    mockChangeDetectorRef = new MockChangeDetectorRef()
+
+    await TestBed.configureTestingModule({
+      imports: [ImportRecordComponent, TranslateModule.forRoot()],
+      providers: [
+        { provide: Router, useValue: mockRouter },
+        { provide: TranslateService, useClass: MockTranslateService },
+        { provide: NotificationsService, useValue: mockNotificationsService },
+        {
+          provide: RecordsRepositoryInterface,
+          useValue: mockRecordsRepository,
+        },
+        { provide: ChangeDetectorRef, useValue: mockChangeDetectorRef },
+
+        ChangeDetectorRef,
```

Comment:
Could you maybe try using ng-mocks here? This might simplify the test file a bit